app/controllers/search_controller.rb

```diff
@@ -1,59 +1,36 @@
 class SearchController < ApplicationController
-	before_filter :if_not_signed_in
+  before_filter :if_not_signed_in
 
-	def index
-		email = params[:search][:email]
+  def index
+    email = params[:search][:email]
 
-		@matching_users = User
+    @matching_users = User.where.not(id: current_user.id).all
+    @matching_users = @matching_users.where(email: email.strip) if email.present?
+  end
 
-		if !email.nil?
-			email.strip!
-			@matching_users = @matching_users.where(email: email)
-		end
-		@matching_users = @matching_users.where.not(id: current_user.id).all
-	end
+  def posts
+    data_type = params[:search][:data_type]
```

Comment:
data_typr isn't hardcoded anywhere but maybe it should be since we make this string comparison in other places in the app